analysis/cross_partition_combiners.py

```diff
@@ -152,3 +152,31 @@ def _multiply_float_dataclasses_field(dataclass, factor: float):
             setattr(dataclass, field.name, value * factor)
         if dataclasses.is_dataclass(value):
             _multiply_float_dataclasses_field(value, factor)
+
+
+def _per_partition_to_cross_partition_utility(
+        per_partition_utility: metrics.PerPartitionUtilityMetrics,
+        dp_metrics: List[pipeline_dp.Metrics],
+        public_partitions: bool) -> metrics.UtilityReport:
+    """Converts per-partition to cross-partition utility metrics."""
+    # Fill partition selection metrics.
+    prob_to_keep = per_partition_utility.partition_selection_probability_to_keep
+    partition_selection_utility = None
+    if not public_partitions:
+        partition_selection_utility = _partition_selection_per_to_cross_partition(
+            prob_to_keep)
+    # Fill metric errors.
+    metric_errors = None
+    if dp_metrics:
+        metric_errors = []
+        assert len(per_partition_utility.metric_errors) == len(dp_metrics)
```

Comment:
We never expect this to happen, right? So maybe add an error message or a code comment that says `This should not happen`?

Reply:
According the [style guide](https://google.github.io/styleguide/pyguide.html#244-decision) "assert is used to ensure internal correctness", i.e. assert False shouldn't happen, no need to write comment about that